### PR TITLE
Fix the incorrect route path in the user edit page.

### PR DESCRIPTION
**Pull request by CaiCandong**
Regression of #26713

After  #26713 , the base path of user edit has been changed to  `/admin/users/{userid}/edit`
## Before

https://github.com/go-gitea/gitea/assets/50507092/5f4a3f64-fe2b-4499-b110-e01c9d87ea19

### CODE REVIEW

**Review by wxiaoguang — APPROVED** [review]

File: routers/web/web.go [hunk 1]
```diff
@@ -575,9 +575,9 @@ func registerRoutes(m *web.Route) {
 			m.Combo("/new").Get(admin.NewUser).Post(web.Bind(forms.AdminCreateUserForm{}), admin.NewUserPost)
 			m.Get("/{userid}", admin.ViewUser)
 			m.Combo("/{userid}/edit").Get(admin.EditUser).Post(web.Bind(forms.AdminEditUserForm{}), admin.EditUserPost)
-			m.Post("/{userid}/delete", admin.DeleteUser)
-			m.Post("/{userid}/avatar", web.Bind(forms.AvatarForm{}), admin.AvatarPost)
-			m.Post("/{userid}/avatar/delete", admin.DeleteAvatar)
+			m.Post("/{userid}/edit/delete", admin.DeleteUser)
```
Comment (lunny) [inline routers/web/web.go]: You should change template link but not the routers I think.

Comment (CaiCandong) [reply routers/web/web.go]: Done [`3beab41` (#27007)](https://github.com/go-gitea/gitea/pull/27007/commits/3beab41b8f5c064babaeb004e18e02d4792cce33)

**Review by wxiaoguang — DISMISSED** [review]
No, do not do this `ctx.Data["Link"] = "/admin/users/" + ctx.Params(":userid")` , it is wrong.

File: routers/web/web.go (hunk 1 above)
Comment (wxiaoguang) [reply routers/web/web.go]: No, do not do this `ctx.Data["Link"] = "/admin/users/" + ctx.Params(":userid")` , it is wrong.

**Review by lunny — APPROVED** [review]

**Review by wxiaoguang — COMMENTED** [review]
It is still not right.

You only have  `BaseUrl` in `EditUser`, but not in `EditUserPost` ....

ps: I really dislike playing such tricks

**Comment by denyskon** [comment]
Honestly, I don't really understand what this PR is trying to do....

**Comment by CaiCandong** [comment]
The `Link` used to splice the route address started out as `/admin/users/{userId}`, but now it becomes `/admin/users/{userId}/edit`. This results in a 404 for requests for user deletion, avatar upload, etc. to the same route.

File: routers/web/admin/users.go [hunk 2]
```diff
@@ -317,6 +317,7 @@ func EditUser(ctx *context.Context) {
 	ctx.Data["DisableGravatar"] = system_model.GetSettingWithCacheBool(ctx, system_model.KeyPictureDisableGravatar,
 		setting.GetDefaultDisableGravatar(),
 	)
+	ctx.Data["BaseUrl"] = "/admin/users/" + ctx.Params(":userid")
```
Comment (puni9869) [inline routers/web/admin/users.go:320]: use fmt.Printf

**Comment by puni9869** [comment]
> It is still not right.
> 
> You only have `BaseUrl` in `EditUser`, but not in `EditUserPost` ....
> 
> ps: I really dislike playing such tricks

I agree with @wxiaoguang  `BaseUrl` should not be override.

**Comment by CaiCandong** [comment]
> > It is still not right.
> > You only have `BaseUrl` in `EditUser`, but not in `EditUserPost` ....
> > ps: I really dislike playing such tricks
> 
> I agree with @wxiaoguang `BaseUrl` should not be override.

At first, he meant that `Link` should not be overridden, and then that `BaseUrl` and `Link` should not be used at the same time. But now I've switched to `BaseUrl` altogether.

**Comment by puni9869** [comment]
IMO you can take a new ctx variable to embed the url.

**Comment by CaiCandong** [comment]
> IMO you can take a new ctx variable to embed the url.

`BaseUrl` is a new context variable actually

**Comment by puni9869** [comment]
I mean `ctx["PathVaribleName"] = fmt.Printf("<pathurl>/%s", variableName)` in the go code.

**Comment by CaiCandong** [comment]
> I mean `ctx["PathVaribleName"] = fmt.Printf("<pathurl>/%s", variableName)` in the go code.
> 
> In the template `{{.PathVaribleName}}/slug`

image

**Comment by puni9869** [comment]
Apologies for the syntax typo. `ctx.Data["PathV...`
`ctx.Data["PathVaribleName"] = fmt.Printf("<pathurl>/%s", variableName)`

**Comment by wxiaoguang** [comment]
> It is still not right.
> 
> You only have `BaseUrl` in `EditUser`, but not in `EditUserPost` ....

Still not fixed.

Actually, the relative URL could also help. If the browser's current URL is `/foo/bar`, then `./xxx` means` /foo/xxx`.

**Comment by CaiCandong** [comment]
> Actually, the relative URL could also help. If the browser's current URL is `/foo/bar`, then `./xxx` means` /foo/xxx`.

Thank you so much for your patient advice, I'll try this method.

**Review by wxiaoguang — APPROVED** [review]
Keep in mind, Gitea has a very unfriendly backend approach: if you put a variable in `Xxx handler (UserEdit)` , then you might also need to put it in `XxxPost handler (UserEditPost)`. Otherwise, if an user-side error occurs when the end user submits the form, the re-rendered page will go wrong.

Or, just like this PR does, get rid of such variables.

File: templates/admin/user/edit.tmpl [hunk 3]
```diff
@@ -4,7 +4,7 @@
 			{{.locale.Tr "admin.users.edit_account"}}
 		</h4>
 		<div class="ui attached segment">
-			<form class="ui form" action="{{.Link}}" method="post">
+			<form class="ui form" action="./edit" method="post">
```
Comment (delvh) [inline templates/admin/user/edit.tmpl:7]: But isn't that the same as
```suggestion
			<form class="ui form" action="edit" method="post">
```
?
Or are you explicitly doing it to mark it as a path?

**Review by delvh — APPROVED** [review]

File: templates/admin/user/edit.tmpl (hunk 3 above)
Comment (CaiCandong) [reply templates/admin/user/edit.tmpl:7]:
```      
m.Combo("/{userid}/edit").Get(admin.EditUser).Post(web.Bind(forms.AdminEditUserForm{}), admin.EditUserPost)
      m.Post("/{userid}/delete", admin.DeleteUser)
      m.Post("/{userid}/avatar", web.Bind(forms.AvatarForm{}), admin.AvatarPost)
      m.Post("/{userid}/avatar/delete", admin.DeleteAvatar)
```
Indicates explicitly that the requested path is in the same directory as the current page.

**Review by puni9869 — APPROVED** [review]

**Comment by CaiCandong** [comment]
> Keep in mind, Gitea has a very unfriendly backend approach: if you put a variable in `Xxx handler (UserEdit)` , then you might also need to put it in `XxxPost handler (UserEditPost)`. Otherwise, if an user-side error occurs when the end user submits the form, the re-rendered page will go wrong.

Thank you so much for your patience and guidance

**Comment by puni9869** [comment]
are we good with PR.
if Y 🚀

**Pull request merged**